### PR TITLE
findWorkDirectory: get the deepest, not the first result

### PR DESCRIPTION
**Pull request by przepompownia**

### CODE REVIEW

**Comment by przepompownia** [comment]
I have `~/.vim/bundle/foo/` plugin that contains `autoload` directory. The original `findWorkDirectory` with `workDirPatterns` returns the home directory because `~/.git` exists. 

It is probably not a good idea to compare deep by length of path string (assuming that one directory is a parent of the other).

This project seems to have no tests! :(

**Comment by przepompownia** [comment]
Independently of the bug: why the directories from `runtimepath` have to be replaced by theirs potential (potential) parent directories? Why not to pass them directly (in contrast to the currently edited file)?

File: src/common/util.ts [hunk 1]
```diff
@@ -83,12 +83,16 @@ export async function findWorkDirectory(
 ): Promise<string> {
   const dirname = path.dirname(filePath)
   let patterns = [].concat(rootPatterns)
+  let dirCandidate = ''
   for (const pattern of patterns) {
     const [err, dir] =  await pcb(findup)(dirname, pattern)
-    if (!err && dir && dir !== '/') {
-      return dir
+    if (!err && dir && dir !== '/' && dir.length > dirCandidate.length) {
```
Comment (iamcco) [inline src/common/util.ts]: It seems it's worth to filter the home directory too.

**Comment by iamcco** [comment]
> Independently of the bug: why the directories from runtimepath have to be replaced by theirs potential (potential) parent directories? Why not to pass them directly (in contrast to the currently edited file)?

Yes, it's better to detect if it's runtimepath or normal vim file path, and only find the workspace directory when edit normal vim file.

File: src/common/util.ts (hunk 1 above)
Comment (przepompownia) [reply src/common/util.ts]: If we will move this filter into `findWorkDirectory` then we probably change the behavior (tests are needed!). Now if for given directory its "working dir" is home dir, then it is skipped. After adding the comparison its "working dir" is it itself. 

In my opinion name "working dir" is confusing and should be changed to "project root" for function, constant and variable names. What do you think about this?

**Comment by przepompownia** [comment]
> > Independently of the bug: why the directories from runtimepath have to be replaced by theirs potential (potential) parent directories? Why not to pass them directly (in contrast to the currently edited file)?
> 
> Yes, it's better to detect if it's runtimepath or normal vim file path, and only find the workspace directory when edit normal vim file.

If you agree with me, we have the topic for the next PR.

File: src/common/util.ts (hunk 1 above)
Comment (iamcco) [reply src/common/util.ts]: Yes, we need test, but i do not have much time to do it.

> In my opinion name "working dir" is confusing and should be changed to "project root" for function, constant and variable names. What do you think about this?

make sense

**Comment by iamcco** [comment]
> If you agree with me, we have the topic for the next PR.

Yes, always welcome

File: src/server/parser.ts [hunk 2]
```diff
@@ -35,13 +35,13 @@ function startIndex() {
   )
 
   scanProcess.on('message', (mess) => {
-    const { data, log } = mess
+    const { data, msglog } = mess
```
Comment (iamcco) [inline src/server/parser.ts]: This change have conflicts with https://github.com/iamcco/vim-language-server/pull/16

Comment (iamcco) [reply src/server/parser.ts]: I have resolve it

**Pull request merged**